pay-api/src/pay_api/models/routing_slip.py

```diff
@@ -62,6 +69,95 @@ def find_by_number(cls, number: str) -> RoutingSlip:
+
+        query = cls._add_receipt_number(query, search_filter)
+
+        query = cls._add_folio_filter(query, search_filter)
+
+        # Add ordering
+        query = query.order_by(RoutingSlip.created_on.desc())
+
+        if not return_all:
+            pagination = query.paginate(per_page=limit, page=page)
+            result, count = pagination.items, pagination.total
+            if max_no_records > 0:
+                count = max_no_records if max_no_records < count else count
+        else:
+            if max_no_records > 0:
+                pagination = query.paginate(per_page=max_no_records, page=1)
+                result, count = pagination.items, max_no_records
+            else:
+                result = query.all()
+                count = len(result)
+
+        return result, count
+
+    @classmethod
+    def _add_date_filter(cls, query, search_filter):
+        # Find start and end dates for folio search
+        created_from: datetime = None
+        created_to: datetime = None
+        if get_str_by_path(search_filter, 'dateFilter/endDate'):
+            created_to = datetime.strptime(get_str_by_path(search_filter, 'dateFilter/endDate'), '%m/%d/%Y')
+        if get_str_by_path(search_filter, 'dateFilter/startDate'):
+            created_from = datetime.strptime(get_str_by_path(search_filter, 'dateFilter/startDate'), '%m/%d/%Y')
+        if get_str_by_path(search_filter, 'weekFilter/index'):
+            created_from, created_to = get_week_start_and_end_date(
+                int(get_str_by_path(search_filter, 'weekFilter/index')))
+        if get_str_by_path(search_filter, 'monthFilter/month') and get_str_by_path(search_filter,
+                                                                                   'monthFilter/year'):
+            # find month
+            month = int(get_str_by_path(search_filter, 'monthFilter/month'))
+            year = int(get_str_by_path(search_filter, 'monthFilter/year'))
+            created_from, created_to = get_first_and_last_dates_of_month(month=month, year=year)
+        # if passed in details
+        if created_to and created_from:
+            # Truncate time for from date and add max time for to date
+            tz_name = current_app.config['LEGISLATIVE_TIMEZONE']
+            tz_local = pytz.timezone(tz_name)
+
+            created_from = created_from.replace(hour=0, minute=0, second=0, microsecond=0).astimezone(tz_local)
+            created_to = created_to.replace(hour=23, minute=59, second=59, microsecond=999999).astimezone(tz_local)
+            query = query.filter(
+                func.timezone(tz_name, func.timezone('UTC', RoutingSlip.created_on)).between(created_from, created_to))
+        return query
+
+    @classmethod
+    def _add_receipt_number(cls, query, search_filter):
+        if receipt_number := search_filter.get('receiptNumber', None):
+            query = query.join(Payment).filter(
```

Comment:
join already exists on the model, can't we use it ?

Reply:
couldnt reuse the model.